lib/ecto/query/builder.ex

```diff
@@ -217,6 +217,11 @@ defmodule Ecto.Query.Builder do
     {expr, params_acc}
   end
 
+  def escape({:=, _, _}, _type, _params_acc, _vars, _env) do
+    error! "match operator is  not supported: `=`. " <>
+           "Instead use the comparison operator: `==`"
```

Comment:
I would prefer to say: "Did you mean to use `==` instead?". That's because i don't want to assume that all cases of `=` is actually meant to be `==`. It is likely... but hard to say for sure across DBs. What do you think?